### ISSUE TITLE
siyuan as a vscode extension

### ISSUE THREAD

**Issue by WangHaiYang874** [siyuan-note/siyuan#2647]
# feature request: make a vscode extension of siyuan
### In what scenarios do you need this function?

vscode is a great text editor (as well as an IDE) with **bunch of powerful extensions**. If SiYuan can be used within vscode, then then search, text editing, can become instantly viable. for example, I can customize bunch of snippets, _enabling vim/emacs keymaps, etc_. 

Another thing I wish to mention that it is better to **separate editing and rendering** in vscode. In this way, editing can be much smoother. And rendering becomes "preview" in another tab. Like the way you edit markdown in vscode. 

I understand that SiYuan can already be used in a chrome. So, it is very feasible to do the same for vscode by using some typescript to build a vscode client that connects to the local SiYuan server. 

To sum up, the benefits of making a vscode extension:
- great functionalities become available. 
- smoother editing experience, when rendering is in a seperate tab.  

### Other information

there are other similar/non-similar note-taking things that I was considering. 

- [gerby](https://github.com/gerby-project): great latex support. Actually, almost full latex support (supports the drawing package tikz) Maybe I should open a new feature request on latex-tikz support. This is critical to me. And as I understand, it can be achieved by just using js [see tikzjax](https://tikzjax.com/) (no latex installation required in the server or the client).  
- [dendron](https://marketplace.visualstudio.com/items?itemName=dendron.dendron): a knowledge base like siyuan, but in vscode. lamer than SiYuan tho. 
- [vscode markdown note](https://marketplace.visualstudio.com/items?itemName=kortina.vscode-markdown-notes) yet another (semi) knowledge base. its functionality is much simpler than siyuan. But it has the advantage of being text only, aka no database whatsoever. 

Thank You!

**Comment by 88250** [comment]
Hello, it may be technically feasible, but if you don't use the editor that comes with VSC, the editing mode and shortcut keys may not be implemented.

In addition, we still have a large number of features/defects to be resolved, so we don’t have time to consider this, sorry.

I closed it first. If you still need to communicate, please feel free to post, thank you.

**Comment by WangHaiYang874** [comment]
You are totally correct. The keymaps and extensions will probably not work with just an simple extension. 

A similar scenario I just checked is that if vim mode is compatible with the vscode/jupyter notebook extension. And unfortuantely, the keymaps cannot work because the vscode-jupyter uses webview for opening `.ipynb` file. I guess there will be similar troubles when making a vscode-extension of siyuan. Let along that Siyuan does not store its file in plain `.md` file but as `.sy`. 

But it is not impossible. The vscode developers will make the vim keymaps compatible with the jupyter extension within months. If you are considering making this extension in the future, probably it is a good idea to see how this is achieved. 

Best regards,
HW
